magic.js

```diff
@@ -1221,6 +1221,41 @@ function uid(sec, cb) {
   }).catch((err) => { return done(err); })
 }
 
+/**
+ * Provides timing safe comparisons of two strings to prevent
+ * timing based attacks. Returns true if the strings are the
+ * same and false if not.
+ * @function
+ * @api public
+ * 
+ * @param {string} input - The first string to check
+ * @param {string} ref - The reference string to check against
+ * @returns {Boolean} - true if they match, false otherwise
+ */
+module.exports.util.timingSafeCompare = (input, ref) => {
```

Comment:
Why aren't we just doing this?: 
```js
module.exports.util.timingSafeCompare = (input, ref) => {
  inputIsString = typeof input === 'string' || input instanceof String;
  refIsString = typeof ref === 'string' || ref instanceof String;
  if (!inputIsString || !refIsString) throw new TypeError('Inputs must be Strings');

  return cnstcomp(Buffer.from(input), Buffer.from(ref))
}
```

Reply:
I think it's because `crypto.timingSafeEqual` requires both inputs to be of the same length.

Reply:
Yup, that is exactly right. If you check for equality up front before running cnstcomp, you end up with a shortened loop that gives away the length of the reference string through timing.

Reply:
Oh I didn't know about the same length restriction 👍